pkg/migrations/op_create_constraint_test.go

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package migrations_test
+
+import (
+	"database/sql"
+	"strings"
+	"testing"
+
+	"github.com/xataio/pgroll/pkg/migrations"
+)
+
+func TestCreateConstraint(t *testing.T) {
+	t.Parallel()
+
+	invalidName := strings.Repeat("x", 64)
+	ExecuteTests(t, TestCases{
+		{
+			name: "create unique constraint on single column",
+			migrations: []migrations.Migration{
+				{
+					Name: "01_add_table",
+					Operations: migrations.Operations{
+						&migrations.OpCreateTable{
+							Name: "users",
+							Columns: []migrations.Column{
+								{
+									Name: "id",
+									Type: "serial",
+									Pk:   ptr(true),
+								},
+								{
+									Name:     "name",
+									Type:     "varchar(255)",
+									Nullable: ptr(false),
+								},
+							},
+						},
+					},
+				},
+				{
+					Name: "02_create_constraint",
+					Operations: migrations.Operations{
+						&migrations.OpCreateConstraint{
+							Name:    "unique_name",
+							Table:   "users",
+							Type:    "unique",
+							Columns: []string{"name"},
+						},
+					},
+				},
+			},
+			afterStart: func(t *testing.T, db *sql.DB, schema string) {
+				// The index has been created on the underlying table.
+				IndexMustExist(t, db, schema, "users", "unique_name")
+			},
+			afterRollback: func(t *testing.T, db *sql.DB, schema string) {
+				// The index has been dropped from the the underlying table.
+				IndexMustNotExist(t, db, schema, "users", "uniue_name")
+			},
+			afterComplete: func(t *testing.T, db *sql.DB, schema string) {
+				// Complete is a no-op.
+			},
```

Comment:
I think these tests should be extended to actually check that the unique constraint is effective on the `users` column in the new schema version but not on the old.

There is an example of how this can look in `op_set_unique_test.go`.